File: recursion/go/recursion1.go
package main

import "fmt"

// first n natural number
func printFirstNNaturalNumber(number int) {
	// negative number
	if number < 1 {
		return
	}

	// base case
	if number == 1 {
		fmt.Print(1, ", ")
		return
	}
	// recursive call
	printFirstNNaturalNumber(number - 1)
	fmt.Print(number, ", ")

}
func main() {
	var number int
	fmt.Scan(&number)
	printFirstNNaturalNumber(number)
}

File: recursion/java/Recursion1.java
import java.util.Scanner;

public class Recursion1{

public static void printFirstNNaturalNumber(int n ){
    // negative number
    if (n < 1) {
        System.out.println("Number can't be negative");
        return;
    }
    if(n == 1) {
        System.out.print(n+", ");
        return;
    }
    printFirstNNaturalNumber(n - 1);
    System.out.print(n+", ");

}

public static Scanner scan = new Scanner(System.in) ;
public static void main(String ... args){
int n = scan.nextInt();
printFirstNNaturalNumber(n);
}
}